src/components/permissions/routes.ts

```diff
@@ -9,17 +12,28 @@ const routeUserPermissions = async (request: any, reply: any) => {
 }
 
 const routeLogin = async (request: any, reply: any) => {
-  const { username, passwordHash } = request.body || { username: '', passwordHash: '' }
+  const { username, password } = request.body || { username: '', password: '' }
+  const { passwordHash } = (await databaseConnect.getUserDataByUsername(username)) || {}
 
-  if (!(await databaseConnect.verifyUser(username, passwordHash)))
-    return reply.send({ success: false })
+  if (!passwordHash) return reply.send({ success: false })
 
-  return reply.send({
+  if (!(await bcrypt.compare(password, passwordHash))) return reply.send({ success: false })
+
+  // Login successful
+  reply.send({
     success: true,
     ...(await getUserInfo(username)),
   })
 }
 
+const routeCreateHash = async (request: any, reply: any) => {
+  const { password } = request.body
+  const hash = await bcrypt.hash(password, saltRounds)
```

Comment:
I thought that saltRounds should be used in conjunction with `genSalt`to generate salt ? And we would store the salt in db ?

I did a post to `http://localhost:8080/create-hash` with "123456" and got "$2b$10$f8sos4kSywTK8HWFvghacO7BLnAg/464EZvoxlZCy9dziQrfzY4xe" which is different to "passwordHash" in insertData

Reply:
Ok never mind, it looks like bcrypt output incorporates both salt and hash

Reply:
Not a required change, but maybe we can add a little comment saying bcrypt output is not just hash, but has metadata like salt

Reply:
Sure. Done 33455925